python/pyspark/sql/tests.py

```diff
@@ -30,6 +30,19 @@
 import functools
 import time
 import datetime
+import traceback
+
+if sys.version_info[:2] <= (2, 6):
```

Comment:
Looks like most of the other tests still have the `<= (2, 6)` check (see python/pyspark/ml/tests.py) so leaving in place.